{"name": "Расчет случайного кейса", "status": "broken", "statusDetails": {"message": "Exception: Элемент не найден", "trace": "self = <pages.sign_in_page.SignInPage object at 0x00000260557423D0>\nlocator = ('css selector', '#root > div > div:nth-child(3) > div > div > div:nth-child(3) > div > div > div > div:nth-child(4) > div > div > div > div > div > ul > li:nth-child(1) > div')\n\n    def click_on_visible_element(self, locator):\n        try:\n>           element = Wait(self.driver, 5).until((ec.visibility_of_element_located(locator)))\n\n..\\pages\\base_page.py:56: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"922fe7d4ab71d02bed571a4454c2be62\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x000002605572FB00>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message:\n\n..\\..\\..\\..\\AppData\\Local\\pypoetry\\Cache\\virtualenvs\\iifa-economics-auto-tests-v-0-3-eUzg4zEH-py3.11\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:95: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_calculation_constructor.TestCalculationConstructor object at 0x000002605570C390>\nsign_in_to_stand = <pages.sign_in_page.SignInPage object at 0x00000260557423D0>\n\n    @allure.title(\"Расчет случайного кейса\")\n    @allure.feature(\"Ядро расчетов, выгрузка отчета\")\n    @allure.story(\"Выбор случайной МК, случайного кейса и расчет с первыми попавшимися макрой и ФЭМ. \"\n                  \"Переход в отчеты и скачивание файла.\")\n    @allure.severity(allure.severity_level.BLOCKER)\n    @allure.description(f\"Тест запустил - {tester_name} \\nСтенд, на котором запускался тест - {current_stand}\")\n    @pytest.mark.calculacion_constructor\n    def test_calculation(self, sign_in_to_stand):\n        with allure.step(\"Прохождение авторизации на стенде\"):\n            page = sign_in_to_stand\n        with allure.step(\"Переход в конструктор расчетов\"):\n            CCPage.move_to_calculation_constructor_page(page)\n            sleep(1)\n            CCPage.is_page_calculation_page(page)\n        with allure.step(\"Расчет с случайным кейсом, первыми попавшимися макрой и ФЭМ\"):\n>           CCPage.calculate_random_case(page)\n\ntest_calculation_constructor.py:64: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\pages\\calculation_constructor_page.py:174: in calculate_random_case\n    self.click_on_visible_element(CCLocators.first_Macro)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <pages.sign_in_page.SignInPage object at 0x00000260557423D0>\nlocator = ('css selector', '#root > div > div:nth-child(3) > div > div > div:nth-child(3) > div > div > div > div:nth-child(4) > div > div > div > div > div > ul > li:nth-child(1) > div')\n\n    def click_on_visible_element(self, locator):\n        try:\n            element = Wait(self.driver, 5).until((ec.visibility_of_element_located(locator)))\n            if not ec.element_to_be_clickable(element):\n                raise Exception(\"Элемент не кликабельный\")\n            return element.click()\n        except TimeoutException:\n>           raise Exception(\"Элемент не найден\")\nE           Exception: Элемент не найден\n\n..\\pages\\base_page.py:61: Exception"}, "description": "Тест запустил - Данил Подольный \nСтенд, на котором запускался тест - http://genrestest.nntc.pro/economics/", "steps": [{"name": "Прохождение авторизации на стенде", "status": "passed", "start": 1676537006091, "stop": 1676537006091}, {"name": "Переход в конструктор расчетов", "status": "passed", "start": 1676537006091, "stop": 1676537007826}, {"name": "Расчет с случайным кейсом, первыми попавшимися макрой и ФЭМ", "status": "broken", "statusDetails": {"message": "Exception: Элемент не найден\n", "trace": "  File \"C:\\Users\\dpodolny\\PycharmProjects\\DatcyEstateAT_v.0.1\\IIFA_Economics_Auto-tests_v.0.3\\tests\\test_calculation_constructor.py\", line 64, in test_calculation\n    CCPage.calculate_random_case(page)\n  File \"C:\\Users\\dpodolny\\PycharmProjects\\DatcyEstateAT_v.0.1\\IIFA_Economics_Auto-tests_v.0.3\\pages\\calculation_constructor_page.py\", line 174, in calculate_random_case\n    self.click_on_visible_element(CCLocators.first_Macro)\n  File \"C:\\Users\\dpodolny\\PycharmProjects\\DatcyEstateAT_v.0.1\\IIFA_Economics_Auto-tests_v.0.3\\pages\\base_page.py\", line 61, in click_on_visible_element\n    raise Exception(\"Элемент не найден\")\n"}, "start": 1676537007826, "stop": 1676537033510}], "attachments": [{"name": "stdout", "source": "4897a9da-4ef6-4234-bf9f-fb5c4fe17494-attachment.txt", "type": "text/plain"}], "start": 1676537006080, "stop": 1676537033515, "uuid": "d68b70f8-7341-47d9-88c9-439f226c9f48", "historyId": "84a8ff3e4ba7bc57ea13d5b0419ccf6f", "testCaseId": "a8bd6ff1bb033a194169651d6e11754b", "fullName": "tests.test_calculation_constructor.TestCalculationConstructor#test_calculation", "labels": [{"name": "story", "value": "Выбор случайной МК, случайного кейса и расчет с первыми попавшимися макрой и ФЭМ. Переход в отчеты и скачивание файла."}, {"name": "severity", "value": "blocker"}, {"name": "feature", "value": "Ядро расчетов, выгрузка отчета"}, {"name": "tag", "value": "calculacion_constructor"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_calculation_constructor"}, {"name": "subSuite", "value": "TestCalculationConstructor"}, {"name": "host", "value": "C100119"}, {"name": "thread", "value": "7172-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_calculation_constructor"}]}